app/src/views/features/rules/RuleEditor/components/Header/ActionButtons/CreateRuleButton/actions/index.js

```diff
@@ -196,6 +196,19 @@ export const validateRule = (rule, dispatch, appMode) => {
           error: "missing from field",
         };
       }
+
+      if (pair.source.operator === GLOBAL_CONSTANTS.RULE_OPERATORS.MATCHES) {
+        const capturingGroupPattern = /\((?!\?:)/g;
+        const hasCapturingGroup = capturingGroupPattern.test(pair.source.value);
```

Comment:
Can use this method too instead
https://github.com/requestly/requestly/blob/b5eb2504636d206d6bc2dc14f4edd5fb1a6c1968/app/src/modules/extension/mv3RuleParser/utils.ts#L29-L32